packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -27,9 +25,10 @@ class ModalBarrier extends StatelessComponent {
         if (dismissable)
           Navigator.of(context).pop();
       },
+      behavior: HitTestBehavior.opaque,
       child: new ConstrainedBox(
         constraints: const BoxConstraints.expand(),
-        child: new DecoratedBox(
+        child: color == null ? null : new DecoratedBox(
```

Comment:
If you pass null here, this won't be opaque to hit testing because it won't paint anything.